python/pyarrow/types.pxi

```diff
@@ -82,6 +84,26 @@ cdef bytes _datatype_to_pep3118(CDataType* type):
             return char
 
 
+cdef void* _as_c_pointer(v, allow_null=False) except *:
+    """
+    Convert a Python object to a raw C pointer.
+
+    Used mainly for the C data interface.
+    Integers are accepted as well as capsule objects with a NULL name.
+    (the latter for compatibility with raw pointers exported by reticulate)
+    """
+    cdef void* c_ptr
+    if isinstance(v, int):
+        c_ptr = <void*> <uintptr_t > v
+    elif PyCapsule_CheckExact(v):
+        c_ptr = PyCapsule_GetPointer(v, NULL)
+    else:
```

Comment:
I don't know, is it actually useful? This is not really a user-facing API. End users will call the higher-level bridge functions.

Reply:
Well if I invoke `export_to_c` and `import_to_c` I end up invoking it. I'm asking just because we supported it in R, so thought it made sense to allow consistent behaviour between the two. In the end it's just one `elif` that doesn't seem to involve much maintenance burden.

Reply:
But in which situation would you get a string in the first place?

Reply:
(in general, accepting more implicit conversions also creates a larger risk of crashes / memory corruption)